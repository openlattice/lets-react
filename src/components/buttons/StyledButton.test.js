import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import StyledButton from './StyledButton';

describe('StyledButton', () => {

  let props;
  let mountedStyledButton;

  const styledButton = () => {
    if (!mountedStyledButton) {
      mountedStyledButton = mount(
        <StyledButton {...props} />
      );
    }
    return mountedStyledButton;
  };

  beforeEach(() => {
    props = {

    };
    mountedStyledButton = undefined;
  });

  it('renders without throwing an error', () => {
    const tree = renderer.create(<StyledButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('always renders a button', () => {
    const buttons = styledButton().find('button');
    expect(buttons).toHaveLength(1);
  });

});
