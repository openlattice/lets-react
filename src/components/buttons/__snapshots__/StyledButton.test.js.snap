// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton renders without throwing an error 1`] = `
.c0 {
  background-color: #fff;
  border-color: #c5d5e5;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  color: #135;
  cursor: pointer;
  font-size: 14px;
  line-height: 18px;
  outline: none;
  padding: 10px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0:hover {
  border-color: #95aabf;
}

.c0:disabled {
  border-color: #d9e3ec;
  color: #a3acb5;
  cursor: not-allowed;
}

<button
  className="c0"
/>
`;
